source/Tests/ApiControllerAnalyzerFixture.cs

```diff
@@ -0,0 +1,242 @@
+﻿using Microsoft.CodeAnalysis.Testing;
+using NUnit.Framework;
+using Octopus.RoslynAnalyzers;
+using System;
+using System.Threading.Tasks;
+using NUnit.Framework.Internal;
+using Verify = Microsoft.CodeAnalysis.CSharp.Testing.NUnit.AnalyzerVerifier<Octopus.RoslynAnalyzers.ApiControllerAnalyzer>;
+
+namespace Tests
+{
+    public class ApiControllerAnalyzerFixture
+    {
+        [Test]
+        public async Task NoDiagnosticsOnWellFormedRequest()
+        {
+            var source = WithOctopusTypes(@"
+namespace Octopus.Core
+{
+  public class GetFooController : ControllerBase {
+    public GetFooResponse GetFoo(GetFooRequest request) => new();
+  }
+}");
+            await Verify.VerifyAnalyzerAsync(source);
+        }
+
+        [Test]
+        public async Task MustNotHaveSwaggerOperationAttribute()
+        {
+            var source = WithOctopusTypes(@"
+namespace Octopus.Core
+{
+  namespace _SwaggerOperation {
```

Comment:
Controllers are annoying because they have to come with matching Request/Response types.
My workaround was just to make the same `GetFoo` to re-use the request/response but use different namespaces to allow multiple controllers